clouddriver-openstack/src/main/groovy/com/netflix/spinnaker/clouddriver/openstack/deploy/description/servergroup/ServerGroupParameters.groovy

```diff
@@ -47,32 +46,36 @@ class ServerGroupParameters {
 
   static final ObjectMapper objectMapper = new ObjectMapper()
 
   Map<String, String> toParamsMap() {
     [
-      flavor              : instanceType,
-      image               : image,
-      max_size            : maxSize ? maxSize.toString() : null,
-      min_size            : minSize ? minSize.toString() : null,
-      desired_size        : desiredSize ? desiredSize.toString() : null,
-      network_id          : networkId,
-      subnet_id           : subnetId,
-      load_balancers      : loadBalancers ? loadBalancers.join(',') : null,
-      security_groups     : securityGroups ? securityGroups.join(',') : null,
-      autoscaling_type    : autoscalingType ? autoscalingType.toString() : null,
-      scaleup_cooldown    : scaleup?.cooldown ? scaleup.cooldown.toString() : null,
-      scaleup_adjustment  : scaleup?.adjustment ? scaleup.adjustment.toString() : null,
-      scaleup_period      : scaleup?.period ? scaleup.period.toString() : null,
-      scaleup_threshold   : scaleup?.threshold ? scaleup.threshold.toString() : null,
-      scaledown_cooldown  : scaledown?.cooldown ? scaledown.cooldown.toString() : null,
-      scaledown_adjustment: scaledown?.adjustment ? scaledown.adjustment.toString() : null,
-      scaledown_period    : scaledown?.period ? scaledown.period.toString() : null,
-      scaledown_threshold : scaledown?.threshold ? scaledown.threshold.toString() : null,
-      tags                : objectMapper.writeValueAsString(tags ?: [:]) ?: null,
-      user_data           : rawUserData ?: null
+      flavor               : instanceType,
+      image                : image,
+      max_size             : maxSize ? maxSize.toString() : null,
+      min_size             : minSize ? minSize.toString() : null,
+      desired_size         : desiredSize ? desiredSize.toString() : null,
+      network_id           : networkId,
+      subnet_id            : subnetId,
+      load_balancers       : loadBalancers ? loadBalancers.join(',') : null,
+      security_groups      : securityGroups ? securityGroups.join(',') : null,
+      autoscaling_type     : autoscalingType ? autoscalingType.toString() : null,
+      scaleup_cooldown     : scaleup?.cooldown ? scaleup.cooldown.toString() : null,
+      scaleup_adjustment   : scaleup?.adjustment ? scaleup.adjustment.toString() : null,
+      scaleup_period       : scaleup?.period ? scaleup.period.toString() : null,
+      scaleup_threshold    : scaleup?.threshold ? scaleup.threshold.toString() : null,
+      scaledown_cooldown   : scaledown?.cooldown ? scaledown.cooldown.toString() : null,
+      scaledown_adjustment : scaledown?.adjustment ? scaledown.adjustment.toString() : null,
+      scaledown_period     : scaledown?.period ? scaledown.period.toString() : null,
+      scaledown_threshold  : scaledown?.threshold ? scaledown.threshold.toString() : null,
+      source_user_data_type: sourceUserDataType ?: null,
+      source_user_data     : sourceUserData ?: null,
```

Comment:
user_data is the converted bytes. source_user_data is merely stored so we can retrieve and show back to the user.